### PR TITLE
update event design doc

### PR DESCRIPTION
**Pull request by kaka89**
Add `how to create producer` and `how to create consumer` section

### CODE REVIEW

File: backend/event/design.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ The schema:
   data; //the real data of the event
 ```
 
-## 3.3 Event publisher
+## 3.3 Event producer
```
Comment (davis999) [inline backend/event/design.md]: above is event publisher, should update above content.

File: backend/event/design.md [hunk 2]
```diff
@@ -49,6 +49,89 @@ The producer will start when the service start, each micro-service constains an
...
+  private String customerId;
+
+  /**
+   * anonymous id.
+   */
+  private String anonymousId;
+```
+### 3.6.2 Define the topics
+Different broker has different way to create topic.
+For now, we use google's pub/sub, so fellow this doc: [google pub/sub](https://cloud.google.com/pubsub/docs/)
```
Comment (davis999) [inline backend/event/design.md]: fellow should be follow.

File: backend/event/design.md [hunk 3]
```diff
@@ -49,6 +49,89 @@ The producer will start when the service start, each micro-service constains an
...
+### 3.6.3 Define the event entity
+When some thing happened, then the somain service produce a event and save it to database.
```
Comment (davis999) [inline backend/event/design.md]: what is the meaning of "somain"? same?

File: backend/event/design.md [hunk 4]
```diff
@@ -49,6 +49,89 @@ The producer will start when the service start, each micro-service constains an
...
+public class EventPublisher {
+  @Scheduled(fixedRate = 200)
+  public void executor() {
+    // do the publish action
+  }
+}
+```
+#### 3.6.5.2 Create the publisher client with `message-client` lib
+Message-client lib is the lib we used to publish message to event bus.
+- first, inport the dependency, see README of `message-client`
```
Comment (davis999) [inline backend/event/design.md:121]: please add a link.

File: backend/event/design.md [hunk 5]
```diff
@@ -64,4 +147,32 @@ When the consumer start, then send a subscribe action to broker. The topic is co
 ## 4.6 Work flow
 - Event reader fetch events from broker.
 - Call processor to process the events.
-![Image](./consumer.png)
\ No newline at end of file
+![Image](./consumer.png)
+
+## 4.7 How to add consumer
+This section shows how to create a consumer in micro service.
+### 4.7.1 Create a subscripber for a topic
```
Comment (davis999) [inline backend/event/design.md]: subscripber should be subscriber

File: backend/event/design.md [hunk 6]
```diff
@@ -49,6 +49,89 @@ The producer will start when the service start, each micro-service constains an
 - After publish success, delete the event from db.
 ![Image](./producer.png)
 
+## 3.6 How to add create producer
```
Comment (davis999) [inline backend/event/design.md]: what is the meaning of "add create"?

File: backend/event/design.md (hunk 4 above)
Comment (kaka89) [reply backend/event/design.md:121]: Not finished yet

File: backend/event/design.md [hunk 7]
```diff
@@ -30,22 +30,22 @@ The schema:
   data; //the real data of the event
 ```
```
Comment (davis999) [inline backend/event/design.md]: add data type.

Comment (kaka89) [reply backend/event/design.md]: done.

File: backend/event/design.md [hunk 8]
```diff
@@ -49,6 +49,89 @@ The producer will start when the service start, each micro-service constains an
...
+As we may got several topic, so we need to create the same number of producer.
+#### 3.6.5.3 publish the event to event bus
+publish the event to event bus, then the producer's work is done.
+- first, fetch the events from db, and change their status to `PENDING`. or fetch those event whose status is `PENDING`, but created long time ago(maybe one minute ago).
+- second, create messages with those events.
+- third, publish those messages to event bus.
+
+
 # 4. Event Consumer
 Event consumer contains two part: `receiver`, `processor`, `manager`
```
Comment (davis999) [inline backend/event/design.md]: THREE parts!!! End with "."

File: backend/event/design.md (hunk 6 above)
Comment (kaka89) [reply backend/event/design.md]: done.

File: backend/event/design.md (hunk 8 above)
Comment (kaka89) [reply backend/event/design.md]: done.

File: backend/event/design.md (hunk 6 above)
Comment (davis999) [reply backend/event/design.md]: "3.4 how to start" does not describe how to start, and i think "3.6 how to create producer" is enough, delete 3.4.

Comment (davis999) [reply backend/event/design.md]: 3.3.1 to 3.3.3 is the workflow of publisher, i think it should be ordered list, not headers.

File: backend/event/design.md [hunk 9]
```diff
@@ -49,6 +49,89 @@ The producer will start when the service start, each micro-service constains an
 - After publish success, delete the event from db.
 ![Image](./producer.png)
 
+## 3.6 How to add create producer
+This section shows how to create a producer in micro service.
+### 3.6.1 Define the event
+Different service produce different events. So the first step is define the events that this service may produced.
```
Comment (davis999) [inline backend/event/design.md]: "is define" should be "is to define", "may produced" should be "may produce".

File: backend/event/design.md [hunk 10]
```diff
@@ -49,6 +49,89 @@ The producer will start when the service start, each micro-service constains an
...
+```java
+Producer producer = DefaultProducerFactory.createGoogleProducer(`projectId`, `topicName`);
+```
+As we may got several topic, so we need to create the same number of producer.
+#### 3.6.5.3 publish the event to event bus
+publish the event to event bus, then the producer's work is done.
+- first, fetch the events from db, and change their status to `PENDING`. or fetch those event whose status is `PENDING`, but created long time ago(maybe one minute ago).
+- second, create messages with those events.
+- third, publish those messages to event bus.
+
```
Comment (davis999) [inline backend/event/design.md]: in 3.5 Workflow, there is a setp "After publish success, delete the event from db", but i can not find the detail description.

File: backend/event/design.md [hunk 11]
```diff
@@ -64,4 +147,32 @@ When the consumer start, then send a subscribe action to broker. The topic is co
 ## 4.6 Work flow
```
Comment (davis999) [inline backend/event/design.md]: 1. 4.1 to 4.3 is simple description, i think a list is ok.
2. "4.4 How to start does" not describe how to start, remove it.
3. "Work flow" should be "Workflow".

File: backend/event/design.md (hunk 9 above)
Comment (kaka89) [reply backend/event/design.md]: both `is define`  and `is to define` are ok.

File: backend/event/design.md (hunk 11 above)
Comment (kaka89) [reply backend/event/design.md]: done.

File: backend/event/design.md (hunk 10 above)
Comment (kaka89) [reply backend/event/design.md]: done.

**Pull request merged**